tests/worker/test_worker.py

```diff
@@ -931,6 +932,29 @@ def state(self):
 #         )
 
 
+async def test_can_run_autoscaling_polling_worker(
```

Comment:
Is there any way we can assert that Core is at least working with what was passed? Maybe via metrics or something?

Reply:
Did it with metrics on startup.

Fun fact: Don't forget that the union discrimination across the pyo3 boundary relies on the variants not having same-named fields. That tripped me up for a while trying to figure out why the config was set properly but not read by core properly, so, test was worth it.